vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,208 @@
+from typing import Set, List, Generator
+
+import re
+import asyncio
+from bs4 import BeautifulSoup
+from packageurl import PackageURL
+import requests
+
+import yaml
+from markdown import markdown
+
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import GitDataSource
+from vulnerabilities.data_source import Reference
+from vulnerabilities.data_source import VulnerabilitySeverity
+from vulnerabilities.severity_systems import scoring_systems
+from vulnerabilities.package_managers import GitHubTagsAPI
+from vulnerabilities.helpers import is_cve
+
+
+REPOSITORY = "mozilla/foundation-security-advisories"
+MFSA_FILENAME_RE = re.compile(r"mfsa(\d{4}-\d{2,3})\.(md|yml)$")
+
+
+class MozillaDataSource(GitDataSource):
+    def __enter__(self):
+        super(MozillaDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, subdir="announce"
+            )
+
+        # Do we need this ?
+        # self.version_api = GitHubTagsAPI()
+        # self.set_api()
+
+    def set_api(self):
+        repository = "/".join(self.config.repository_url.split("/")[-2:])
+        asyncio.run(self.version_api.load_api([repository]))
+
+    def added_advisories(self) -> Set[Advisory]:
+        return self._load_advisories(self._added_files)
+
+    def updated_advisories(self) -> Set[Advisory]:
+        return self._load_advisories(self._updated_files)
+
+    def _load_advisories(self, files) -> Set[Advisory]:
+        """
+        Yields list of advisories of batch size
+        """
+        files = [
+            f for f in files if f.endswith(".md") or f.endswith(".yml")
+        ]  # skip irrelevant files
+
+        advisories = []
+        for path in files:
+            for advisory in self._to_advisories(path):
+                advisories.append(advisory)
+                if len(advisories) >= self.batch_size:
+                    yield advisories
+                    advisories = []
+
+        # In case batch size is too high
```

Comment:
Please have a look at the above lines
```py
        advisories = []
        for path in files:
            for advisory in self._to_advisories(path):
                advisories.append(advisory)
                if len(advisories) >= self.batch_size:
                    yield advisories
                    advisories = []
```
It only yields `if len(advisories) >= self.batch_size:` which would never be true in case the batch size is greater than the total number of advisories present. The loop would just complete and reach the end of the function. So, I've added a final yield.